emscripten.py

```diff
@@ -2769,11 +2769,8 @@ def normalize_line_endings(text):
   return text
 
 
-def run(infile, outfile, memfile):
-  temp_files = get_configuration().get_temp_files()
-  infile, outfile = substitute_response_files([infile, outfile])
-
-  if not shared.Settings.BOOTSTRAPPING_STRUCT_INFO:
+def generate_struct_info():
+  if not shared.Settings.STRUCT_INFO and not shared.Settings.BOOTSTRAPPING_STRUCT_INFO:
```

Comment:
why has STRUCT_INFO been added as a check here?

Reply:
This avoids generating the struct into twice.

In the old code we would do:

```
if not BOOTSTRAPPING_STRUCT_INFO:
   STRUCT_INFO = ..do work ..;
```

However in the new layout we might end up calling this code twice so I added the extra guard here which basically says "if we have already got the struct info from the cache then don't bother getting it again".  Perhaps we should call this function `load_struct_info` since it doesn't always generate it.